packages/playwright/src/runner/projectUtils.ts

```diff
@@ -24,26 +24,31 @@ import { createFileMatcher } from '../util';
 const readFileAsync = promisify(fs.readFile);
 const readDirAsync = promisify(fs.readdir);
 
+// The difference to forceRegExp is that we want to match the whole string.
+function forceBoundedRegExp(pattern: string): RegExp {
+  const match = pattern.match(/^\/(.*)\/([gi]*)$/);
+  if (match)
+    return new RegExp(match[1], match[2]);
+  return new RegExp(`^${pattern}$`, 'gi');
+}
+
 export function filterProjects(projects: FullProjectInternal[], projectNames?: string[]): FullProjectInternal[] {
   if (!projectNames)
     return [...projects];
-  const projectsToFind = new Set<string>();
-  const unknownProjects = new Map<string, string>();
-  projectNames.forEach(n => {
-    const name = n.toLocaleLowerCase();
-    projectsToFind.add(name);
-    unknownProjects.set(name, n);
-  });
+  const unmatchedProjectFilters = new Set<string>(projectNames);
   const result = projects.filter(project => {
-    const name = project.project.name.toLocaleLowerCase();
-    unknownProjects.delete(name);
-    return projectsToFind.has(name);
+    for (const projectName of projectNames) {
+      if (forceBoundedRegExp(projectName).test(project.project.name)) {
+        unmatchedProjectFilters.delete(projectName);
+        return true;
+      }
+    }
```

Comment:
nit: return false in the end? I'd also do `unmatchedProjectFilters.add(projectName);` there instead of deleting from the set.

Reply:
Reversing unmatchedProjectFilters moves the same complexity to the end, keeping it for now.